src/main/scala/net/snowflake/spark/snowflake/pushdowns/SnowflakePlan.scala

```diff
@@ -10,6 +10,8 @@ import org.apache.spark.sql.types.{StructField, StructType}
 case class SnowflakePlan(output: Seq[Attribute], rdd: RDD[InternalRow])
     extends SparkPlan {
 
+  override protected def withNewChildrenInternal(newChildren: IndexedSeq[SparkPlan]): SparkPlan = this
+
```

Comment:
Why `withNewChildrenInternal` is added?

Reply:
It is a new abstract function introduced in TreeNode, which is extended by this class. see more on my previous comment.

Reply:
If this PR need to be merged into `master`, we have to figure out what's its use case and implement a reasonable one.

Reply:
It is an optimization for `TreeNode#mapChildren`. This function doeas a check `containsChild.nonEmpty` which is allways `false` in case of `SnowflakePlan`, because there is:

```scala
override def children: Seq[SparkPlan] = Nil
```

so this function is never called, I just made it return `this`, as suggested in https://issues.apache.org/jira/browse/SPARK-34989. This can just throw an exception, since it never gets called for a childless node.